pkg/controller/machine/machine_controller.go

```diff
@@ -305,3 +354,27 @@ func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cl
 
 	return corev1Remote.Nodes().Delete(name, &metav1.DeleteOptions{})
 }
+
+// getMachines returns all of the Machine objects that belong to the same
+// cluster as the provided Machine
+func (r *ReconcileMachine) getMachines(ctx context.Context, machine *clusterv1.Machine) ([]*clusterv1.Machine, error) {
```

Comment:
We could pass Cluster in here and rename this method to something like `getMachinesForCluster`

Reply:
True, but it doesn't really feel strictly necessary. If you feel strongly, I'll make the change.

Reply:
It will also require additional refactoring for other functions and be anti-pattern to the other functions that simply accept a machine. I think this should stay as is.

Reply:
Nah it's fine, it feels weird to get machines from a Machine, but we can refactor later

Reply:
Hi @vincepri,

There's a function for getting the cluster for a machine, but that would just involve an additional, unnecessary round-trip to the API server. This seems more efficient.

Reply:
We could change this to `getMachinesInCluster(namespace, name string) ([]*clusterv1.Machine, error)` and make the caller pass in the cluster namespace and name. WDYT?

Reply:
Honestly? I think this is a bug that causes a pretty bad failure, and I’d like to see the fix merged sooner than later. If possible, I’d prefer to revisit this later.